aodncore/pipeline/steps/harvest.py

```diff
@@ -218,7 +219,11 @@ def run_additions(self, harvester_map, tmp_base_dir):
                 try:
                     self.execute_talend(self._config.trigger_config[harvester]['exec'], matched_files, talend_base_dir)
                 except Exception:
-                    self.undo_processed_files()
+                    undo_map = [{harvester: matched_files}]
+                    if self.undo_previous_slices:
+                        undo_map.extend(self.harvested_file_map)
```

Comment:
Humph! I just realised this is not quite right... if `undo_previous_slices` is True, it's all good, it will undo everything. But if it's False, it will only undo the failing harvest, _but not any previous harvests on the current slice_!

Reply:
Anyway, the default behaviour (True) is good, so let's go ahead and test that in RC. We can fix up the other option (currently not required) later.

Reply:
Is that as a result of testing? I don't think that's the case, because the harvested_file_map is updated inside the loop, so on a _per-harvester_ basis as soon as it completes successfully. The previous harvests for the current slice should be in the map, and therefore undone. Have you found otherwise from testing?

Reply:
I think you're right that this could be a future enhancement. We could cause the behaviour to differ based on different harvest_params, but should scope that out first.